service/src/main/java/datawave/microservice/dictionary/DataDictionaryWebController.java

```diff
@@ -0,0 +1,275 @@
+                    ResponseObjectFactory<DESC,DICT,META,FIELD,FIELDS> responseObjectFactory, AccumuloConnectionService accumloConnectionService) {
+        this.dataDictionaryConfiguration = dataDictionaryConfiguration;
+        this.dataDictionary = dataDictionary;
+        this.responseObjectFactory = responseObjectFactory;
+        this.accumuloConnectionService = accumloConnectionService;
+        dataDictionary.setNormalizationMap(dataDictionaryConfiguration.getNormalizerMap());
+    }
+    
+    /**
+     * Returns the DataDictionary for the given parameters.
+     *
+     * @param modelName
+     *            Optional model name
+     * @param modelTableName
+     *            Optional model table name
+     * @param metadataTableName
+     *            Optional metadata table name
+     * @param queryAuthorizations
+     *            Optional query authorizations
+     * @param dataTypeFilters
+     *            Optional data type filters
+     * @param currentUser
+     *            the current user
+     * @return The ModelAndView for datadictionary.html which contains the data dictionary fields.
+     * @throws Exception
+     *             if there is any problem fetching the entries
+     */
+    @GetMapping("/")
+    @Timed(name = "dw.dictionary.data.get", absolute = true)
+    public ModelAndView get(@RequestParam(required = false) String modelName, @RequestParam(required = false) String modelTableName,
+                    @RequestParam(required = false) String metadataTableName, @RequestParam(name = "auths", required = false) String queryAuthorizations,
+                    @RequestParam(defaultValue = "") String dataTypeFilters, @AuthenticationPrincipal ProxiedUserDetails currentUser) throws Exception {
+        Connection connection = accumuloConnectionService.getConnection(metadataTableName, modelTableName, modelName, currentUser);
+        // If the user provides authorizations, intersect it with their actual authorizations
+        connection.setAuths(accumuloConnectionService.getDowngradedAuthorizations(queryAuthorizations, currentUser));
+        
+        Collection<String> dataTypes = (StringUtils.isBlank(dataTypeFilters) ? Collections.emptyList() : Arrays.asList(dataTypeFilters.split(",")));
+        
+        Collection<META> fields = dataDictionary.getFields(connection, dataTypes, dataDictionaryConfiguration.getNumThreads());
+        DICT dataDictionary = responseObjectFactory.getDataDictionary();
+        dataDictionary.setFields(fields);
+        // Ensure that empty internal field names will be set to the field name instead.
+        dataDictionary.transformFields(TRANSFORM_EMPTY_INTERNAL_FIELD_NAMES);
+        
+        ModelAndView mav = new ModelAndView();
+        mav.setViewName("datadictionary");
+        mav.addObject("jqueryUri", dataDictionary.getJqueryUri() + "jquery.min.js");
+        mav.addObject("datatablesUri", dataDictionary.getDatatablesUri() + "jquery.dataTables.min.js");
```

Comment:
Ahh very nice. Thanks! I think it would probably be cleaner to do it your way with accessing it directly in thymeleaf so we aren't unnecessarily passing around these strings (also removes the need for a couple getters I made). Let me know if I should change this or keep what I have now.